### PR TITLE
Update scripts to enable rest function call only for node #5 and #6

### PR DESCRIPTION
**Pull request by ehgmsdk20**
Changes:
- updated scripts for function call server node

Related issues:

- #660

### CODE REVIEW

File: deploy_blockchain_genesis_gcp.sh [hunk 1]
```diff
@@ -225,10 +225,10 @@ printf "\n\n##########################\n# Starting parent node 4 #\n############
 gcloud compute ssh $NODE_4_TARGET_ADDR --command "$START_NODE_COMMAND_BASE 0 4 $ACCOUNT_INJECTION_OPTION" --project $PROJECT_ID --zone $NODE_4_ZONE
 inject_account "4"
 printf "\n\n##########################\n# Starting parent node 5 #\n##########################\n\n"
-gcloud compute ssh $NODE_5_TARGET_ADDR --command "$START_NODE_COMMAND_BASE 0 5 $ACCOUNT_INJECTION_OPTION" --project $PROJECT_ID --zone $NODE_5_ZONE
+gcloud compute ssh $NODE_5_TARGET_ADDR --command "$START_NODE_COMMAND_BASE 0 5 $ACCOUNT_INJECTION_OPTION --restfunc" --project $PROJECT_ID --zone $NODE_5_ZONE
```
Comment (platfowner) [inline deploy_blockchain_genesis_gcp.sh:228]: what about ---rest-func like --keep-code?

Comment (ehgmsdk20) [reply deploy_blockchain_genesis_gcp.sh]: it looks good! i edited. PTAL

**Comment by platfowner** [comment]
Could you update the PR title with a more relative one?

File: start_node_genesis_gcp.sh [hunk 2]
```diff
@@ -32,15 +34,20 @@ function parse_options() {
 # Parse options.
 KEEP_CODE_OPTION=""
 ACCOUNT_INJECTION_OPTION=""
-if [[ "$#" -gt 3 ]]; then
-    parse_options "$4"
-    if [[ "$#" = 5 ]]; then
-        parse_options "$5"
-    fi
-fi
+ENABLE_REST_FUNCTION_CALL="false"
```
[the remark this answers is not in the document]
Comment (ehgmsdk20) [reply start_node_genesis_gcp.sh]: I edited it, but I don't know if I understood your intention clearly.

File: deploy_blockchain_incremental_gcp.sh [hunk 3]
```diff
@@ -178,6 +178,9 @@ function deploy_node() {
     # 2. Start node
     printf "\n\n[[[[ Starting node $node_index ]]]]\n\n"
     START_CMD="gcloud compute ssh $node_target_addr --command '. start_node_incremental_gcp.sh $SEASON 0 $node_index $SYNC_MODE $ACCOUNT_INJECTION_OPTION' --project $PROJECT_ID --zone $node_zone"
+    if [[ $node_index -gt 4 ]]; then
+        START_CMD="gcloud compute ssh $node_target_addr --command '. start_node_incremental_gcp.sh $SEASON 0 $node_index $SYNC_MODE $ACCOUNT_INJECTION_OPTION --restfunc' --project $PROJECT_ID --zone $node_zone"
```
[the remark this answers is not in the document]
Comment (ehgmsdk20) [reply deploy_blockchain_incremental_gcp.sh]: it looks more readable! thanks

File: start_node_incremental_gcp.sh [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
-if [[ "$#" -lt 4 ]] || [[ "$#" -gt 5 ]]; then
-    printf "Usage: bash start_node_incremental_gcp.sh [dev|staging|spring|summer] <Shard Index> <Node Index> [fast|full] [--keystore|--mnemonic]\n"
+if [[ "$#" -lt 4 ]] || [[ "$#" -gt 6 ]]; then
+    printf "Usage: bash start_node_incremental_gcp.sh [dev|staging|spring|summer] <Shard Index> <Node Index> [fast|full] [--keystore|--mnemonic] [--restfunc]\n"
```
[the remark this answers is not in the document]
Comment (ehgmsdk20) [reply start_node_incremental_gcp.sh]: Done!

File: start_node_incremental_gcp.sh [hunk 5]
```diff
@@ -98,9 +98,43 @@ fi
 
 export SYNC_MODE="$4"
 printf "SYNC_MODE=$SYNC_MODE\n"
-ACCOUNT_INJECTION_OPTION=$5
+
+function parse_options() {
+    local option="$1"
+    if [[ "$option" = '--restfunc' ]]; then
+        ENABLE_REST_FUNCTION_CALL="true"
```
[the remark this answers is not in the document]
Comment (ehgmsdk20) [reply start_node_incremental_gcp.sh]: Done! PTAL

File: deploy_blockchain_genesis_gcp.sh [hunk 6]
```diff
@@ -209,27 +209,22 @@ fi
...
-inject_account "4"
-printf "\n\n##########################\n# Starting parent node 5 #\n##########################\n\n"
-gcloud compute ssh $NODE_5_TARGET_ADDR --command "$START_NODE_COMMAND_BASE 0 5 $ACCOUNT_INJECTION_OPTION" --project $PROJECT_ID --zone $NODE_5_ZONE
-inject_account "5"
-printf "\n\n##########################\n# Starting parent node 6 #\n##########################\n\n"
-gcloud compute ssh $NODE_6_TARGET_ADDR --command "$START_NODE_COMMAND_BASE 0 6 $ACCOUNT_INJECTION_OPTION" --project $PROJECT_ID --zone $NODE_6_ZONE
-inject_account "6"
+NUM_NODES=7
+index=0
+while [ $index -lt $NUM_NODES ]
```
Comment (platfowner) [inline deploy_blockchain_genesis_gcp.sh:214]: nice refactoring, thanks!

File: start_node_incremental_gcp.sh [hunk 7]
```diff
@@ -98,9 +98,47 @@ fi
...
+    else
+        echo "Invalid option: $option"
+        exit
+    fi
+}
+
+ACCOUNT_INJECTION_OPTION=""
+REST_FUNC_OPTION=""
+
+if [[ "$#" -gt 4 ]]; then
```
Comment (platfowner) [inline start_node_incremental_gcp.sh:127]: minor: what about using the while loop used in in start_node_genesis_gcp.sh?

**Review by platfowner — APPROVED** [review]
LGTM with some minor comments.
Thanks for the nice implementation!

File: start_node_incremental_gcp.sh (hunk 7 above)
Comment (ehgmsdk20) [reply start_node_incremental_gcp.sh]: I edited, thanks!

**Pull request merged**